### DISCUSSION TITLE
Crud Resources

### DISCUSSION THREAD

**Discussion by wsdnathan** [orchidsoftware/platform#1584] [Q&A] [answered]
First off the new Resources are great, I like them a bit more than Nova. I still haven't completely figured out Screens, but Resources helped me get working with Orchid a lot easier. Thanks for adding this.

Here is my feedback and questions after using Orchid for a short period of time:

The sort orders for the Resources sidebar navigation seem a little buggy to me. **I wish by default for Resources it would just make them alphabetical.** Depending where it was deployed I was having different sort order displaying. When I tried setting the weight then I ran into even more issues with grouping of my other Orchid routes. The function seems to expect a string and the 2000 integer provided when used pretty much means you have to manually override sort weight on everything for it to render correctly. It may have additionally been conflicting with displayInNavigation().

**Is there some way to customize or add additional commandBar() for Resources?** I couldn't figure it out. Could a method() also just be defined in the same class just as they are with Screens?

Speaking of Screens I tried creating an export method and ran into a bunch of machine code getting spit out instead of XLS file downloading. Is this maybe because of Orchid view SPA? **How would you provide a method that allows for sending headers for a file download,** perhaps just with Link::href() but to where? You can hook methods to Screens but that was the part that didn't work for me, not sure where else to hook in the equivalent of controller actions.

I wish that I could provide export and settings modal methods commandBar() to Resources. Then I feel like I would hardly need to use Screens at all (for any sort of basic data management anyway). Resources are very useful compared to the complexity of Screens and Layouts.

Hope to see continued development of Orchid because I think the way it's set up is smart. I'm surprised I've never heard of this project until recently.

**Comment by tabuna** [comment, marked as answer]
> Is there some way to customize or add additional commandBar() for Resources? I couldn't figure it out. Could a method() also just be defined in the same class just as they are with Screens?

Hi, I am planning to add this after I make a simple post view page (no editing). Now you can use [Actions](https://github.com/orchidsoftware/crud#actions) from the list page


> Speaking of Screens I tried creating an export method and ran into a bunch of machine code getting spit out instead of XLS file downloading. Is this maybe because of Orchid view SPA? How would you provide a method that allows for sending headers for a file download, perhaps just with Link::href() but to where? You can hook methods to Screens but that was the part that didn't work for me, not sure where else to hook in the equivalent of controller actions.

If you want to download a file that is generated automatically, then just return it in the screen method. And add `->rawClick()` to the call button so that there is no SPA processing. For instance:

```php
Button::make('download')
    ->method('exportXLS')
    ->rawClick()
```

> Hope to see continued development of Orchid because I think the way it's set up is smart. I'm surprised I've never heard of this project until recently.

Share your find with others;)

**Reply by bilogic** [reply]
@tabuna I was trying to update this into the documentation and had some thoughts.
Apart from download, are there other use cases for `rawClick()`? JS etc?
If not, is it better to rename it `asDownload()` to make it more readable?

```PHP
Button::make('download')
    ->method('exportXLS')
    ->asDownload()
```
